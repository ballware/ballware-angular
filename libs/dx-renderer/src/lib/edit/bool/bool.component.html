<ballware-edit-fieldset [layoutItem]="layoutItem">
  <dx-check-box *ngIf="layoutItem && layoutItem.options && layoutItem.options.dataMember"
    [name]="layoutItem.options.dataMember"
    [(value)]="value"
    [readOnly]="(readonly$ | async) ?? false">
    <dx-validator>
      <dxi-validation-rule *ngIf="(required$ | async) ?? false" type="required"></dxi-validation-rule>
    </dx-validator>
  </dx-check-box>
</ballware-edit-fieldset>
