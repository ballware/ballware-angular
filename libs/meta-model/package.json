{
  "name": "@ballware/meta-model",
  "version": "8.0.1",
  "type": "commonjs"
}
