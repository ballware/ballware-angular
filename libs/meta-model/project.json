{
  "name": "meta-model",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/meta-model/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/js:tsc",
      "outputs": [
        "{options.outputPath}"
      ],
      "options": {
        "outputPath": "dist/libs/meta-model",
        "main": "libs/meta-model/src/index.ts",
        "tsConfig": "libs/meta-model/tsconfig.lib.json",
        "assets": [
          "libs/meta-model/*.md"
        ]
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "node tools/scripts/publish.mjs meta-model {args.ver} {args.tag}"
      },
      "dependsOn": [
        "build"
      ]
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": [
        "{options.outputFile}"
      ],
      "options": {
        "lintFilePatterns": [
          "libs/meta-model/**/*.ts"
        ]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": [
        "{workspaceRoot}/coverage/{projectRoot}"
      ],
      "options": {
        "jestConfig": "libs/meta-model/jest.config.ts",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
