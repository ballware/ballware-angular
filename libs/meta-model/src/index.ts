export * from './lib/cruditem';
export * from './lib/document';
export * from './lib/entity';
export * from './lib/page';
export * from './lib/queryparams';
export * from './lib/rights';
export * from './lib/scriptutil';
export * from './lib/statistic';
export * from './lib/template';
export * from './lib/tenant';

