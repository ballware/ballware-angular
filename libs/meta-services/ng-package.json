{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../dist/libs/meta-services",
  "lib": {
    "entryFile": "src/index.ts"
  }
}
