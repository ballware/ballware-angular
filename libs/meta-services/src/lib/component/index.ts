import { StoreModule } from '@ngrx/store';
import { componentReducer } from './component.reducer';
import { componentFeatureKey } from './component.state';

export * from './component.actions';
export * from './component.state';

export const ComponentFeatureModule = StoreModule.forFeature(componentFeatureKey, componentReducer);