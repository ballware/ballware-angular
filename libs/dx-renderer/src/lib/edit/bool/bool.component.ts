import { Component, Inject, Input, OnInit } from '@angular/core';
import { EditLayoutItem } from '@ballware/meta-model';
import { EDIT_SERVICE, EditItemRef, EditService } from '@ballware/meta-services';
import { takeUntil } from 'rxjs';
import { WithDestroy } from '../../utils/withdestroy';
import { WithEditItemLifecycle } from '../../utils/withedititemlivecycle';
import { WithReadonly } from '../../utils/withreadonly';
import { WithRequired } from '../../utils/withrequired';
import { WithValidation } from '../../utils/withvalidation';
import { WithValue } from '../../utils/withvalue';
import { WithVisible } from '../../utils/withvisible';

@Component({
  selector: 'ballware-edit-bool',
  templateUrl: './bool.component.html',
  styleUrls: ['./bool.component.scss']
})
export class EditLayoutBoolComponent extends WithVisible(WithRequired(WithValidation(WithReadonly(WithValue(WithEditItemLifecycle(WithDestroy()), () => false as boolean|null|undefined))))) implements OnInit, EditItemRef {

  @Input() initialLayoutItem?: EditLayoutItem;

  public layoutItem: EditLayoutItem|undefined;

  constructor(
    @Inject(EDIT_SERVICE) private editService: EditService) {
    super();
  }

  ngOnInit(): void {
    if (this.initialLayoutItem) {
      this.initLifecycle(this.initialLayoutItem, this.editService, this);

      this.preparedLayoutItem$
        .pipe(takeUntil(this.destroy$))
        .subscribe((layoutItem) => {
          if (layoutItem) {
            this.initValue(layoutItem, this.editService);
            this.initReadonly(layoutItem, this.editService);
            this.initValidation(layoutItem, this.editService);
            this.initRequired(layoutItem, this.editService);
            this.initVisible(layoutItem);

            this.layoutItem = layoutItem;
          }
        });
    }
  }

  public getOption(option: string): any {
    switch (option) {
      case 'value':
        return this.value;
      case 'required':
        return this.required$.getValue();
      case 'readonly':
        return this.readonly$.getValue();
      case 'visible':
        return this.visible$.getValue();        
    }

    return undefined;
  }

  public setOption(option: string, value: unknown) {
    switch (option) {
      case 'value':
        this.setValueWithoutNotification(value as boolean);
        break;
      case 'required':
        this.setRequired(value as boolean);
        break;
      case 'readonly':
        this.setReadonly(value as boolean)
        break;
      case 'visible':
        this.setVisible(value as boolean);
        break;
    }
  }
}
