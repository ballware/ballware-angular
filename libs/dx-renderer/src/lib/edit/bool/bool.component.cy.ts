import { MountConfig, mount } from 'cypress/angular';
import { EditLayoutBoolComponent } from './bool.component';

describe(EditLayoutBoolComponent.name, () => {
  const config: MountConfig<EditLayoutBoolComponent> = {
    declarations: [],
    imports: [],
    providers: []
  }

  it('renders', () => {
     mount(EditLayoutBoolComponent, config);
  })
})
