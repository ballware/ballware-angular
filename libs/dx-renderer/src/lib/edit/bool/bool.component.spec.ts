import { ComponentFixture, TestBed } from '@angular/core/testing';

import { Mock } from 'moq.ts';

import { EditLayoutBoolComponent } from './bool.component';
import { Provider } from '@angular/core';
import { EDIT_SERVICE, EditService } from '@ballware/meta-services';
import { BehaviorSubject } from 'rxjs';
import { EditLayoutItem } from '@ballware/meta-model';

describe('EditLayoutBoolComponent', () => {
  let component: EditLayoutBoolComponent;
  let fixture: ComponentFixture<EditLayoutBoolComponent>;
  
  const mockedEditService = new Mock<EditService>();
        
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ EditLayoutBoolComponent ],
      providers: [        
        {
          provide: EDIT_SERVICE,
          useFactory: () => mockedEditService.object()
        } as Provider         
      ]
    })
    .compileComponents();
  });

  it('should create with lifecycle', () => {
    fixture = TestBed.createComponent(EditLayoutBoolComponent);

    const editorPreparing = jest.fn();
    const editorInitialized = jest.fn();

    mockedEditService
        .setup(instance => instance.editorPreparing$).returns(new BehaviorSubject(editorPreparing))
        .setup(instance => instance.editorInitialized$).returns(new BehaviorSubject(editorInitialized));

    const layoutItem = {
        options: {
            dataMember: 'mockedmember'
        }
    } as EditLayoutItem;
    
    component = fixture.componentInstance;   
    expect(component).toBeTruthy();

    component.initialLayoutItem = layoutItem;
    fixture.detectChanges();
       
    expect(editorPreparing).toBeCalledTimes(1);
    expect(editorPreparing).toBeCalledWith(expect.objectContaining({ dataMember: 'mockedmember' }));

    expect(editorInitialized).toBeCalledTimes(1);
    expect(editorInitialized).toBeCalledWith(expect.objectContaining({ dataMember: 'mockedmember' }));
    
  });
});
