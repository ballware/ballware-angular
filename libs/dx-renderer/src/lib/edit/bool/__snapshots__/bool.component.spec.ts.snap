// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditLayoutBoolComponent should create 1`] = `
<ballware-edit-bool
  _optionRegistry={[Function Array]}
  currentValue$={[Function BehaviorSubject]}
  destroy$={[Function Subject]}
  editService={[Function ]}
  editorEntered$={[Function Subject]}
  editorEvent$={[Function Subject]}
  emailValidation$={[Function BehaviorSubject]}
  getOption={[Function Function]}
  globalReadonly="false"
  initialLayoutItem={[Function Object]}
  layoutItem={[Function Object]}
  notifyValueChange$={[Function Subject]}
  onEntered={[Function Function]}
  onEvent={[Function Function]}
  preparedLayoutItem$={[Function BehaviorSubject]}
  readonly$={[Function BehaviorSubject]}
  refreshValueTrigger$={[Function Subject]}
  registerOption={[Function Function]}
  required$={[Function BehaviorSubject]}
  requiredValidation$={[Function BehaviorSubject]}
  setOption={[Function Function]}
  validationRules$={[Function Observable]}
  visible$={[Function BehaviorSubject]}
>
  <div
    class="dx-texteditor-label mt-2"
    style="position: relative;"
  >
  </div><dx-check-box
    aria-checked="false"
    aria-invalid="false"
    aria-readonly="false"
    class="dx-widget dx-checkbox pt-1 dx-validator dx-visibility-change-handler"
    role="checkbox"
    tabindex="0"
  >
    <input
      name="mockedmember"
      type="hidden"
      value="false"
    />
    <div
      class="dx-checkbox-container"
    >
      <span
        class="dx-checkbox-icon"
      />
    </div>
  </dx-check-box>
</ballware-edit-bool>
`;
